src/debug.rs

```diff
@@ -0,0 +1,7 @@
+#[macro_export]
+macro_rules! debug {
+  ($fmt_str:literal $(, $args:expr)*) => {
+    #[cfg(feature = "debug")]
+    eprintln!($fmt_str, $(,$args)*);
```

Comment:
It might be a better option to add logging, and make this a `trace`?

Reply:
Sure, do you have a preferred [logging frontend crate](https://crates.io/crates/log#in-executables)?

Reply:
I normally just use `log`.  The consumer of the library can choose to use an actual logger to emit the logs to console / etc.